src/wasi.rs

```diff
@@ -0,0 +1,22 @@
+// TODO: flesh this out once actual WASI APIs for this is designed.
+//
+// For now here are just enough definitions to make things *compile*.
+
+use std::path::PathBuf;
+
+pub fn home_dir()       -> Option<PathBuf> { None }
+pub fn cache_dir()      -> Option<PathBuf> { None }
+pub fn config_dir()     -> Option<PathBuf> { None }
+pub fn data_dir()       -> Option<PathBuf> { None }
+pub fn data_local_dir() -> Option<PathBuf> { None }
+pub fn runtime_dir()    -> Option<PathBuf> { None }
+pub fn executable_dir() -> Option<PathBuf> { None }
+pub fn audio_dir()      -> Option<PathBuf> { home_dir().map(|h| h.join("Music")) }
```

Comment:
Can you also use `None` for these?

Reply:
Yeah, sure. I just thought these to be sensible defaults though, of course not the case if the final API actually included specific support for such personal and interactive uses.

Reply:
They all end up as `None`, because `home_dir` returns `None`.

Reply:
I sometimes do things like this at work, since multiple people get to work on different parts of the same workflow, if someone finished their part the dataflow would automatically connect.

It's correct to assume nothing before platform stabilization, though.